src/amdinfer/workers/xmodel.cpp

```diff
@@ -182,8 +185,15 @@ BatchPtr XModel::doRun(Batch* batch, const MemoryPool* pool) {
 #endif  // AMDINFER_ENABLE_LOGGING
 
   std::vector<vart::TensorBuffer*> inputs_ptr;
-  for (const auto& buffer : input_buffers) {
+  for (unsigned i = 0; i < input_buffers.size(); i++) {
+    const auto& buffer = input_buffers[i];
     auto* vart = dynamic_cast<VartTensorBuffer*>(buffer.get());
+    // // Check if model tensor name equals input tensor name
+    // const auto* runner_buffer = runner_->get_input_tensors()[i];
```

Comment:
Having this check is good. Uncomment this and compare the incoming tensor names against metadata_. On error, you need to iterate through the requests of the batch and return an error instead of assert. See [this](https://github.com/Xilinx/inference-server/blob/main/src/amdinfer/workers/migraphx.cpp#L489-L494) in migraphx

Reply:
done